allure-generator/src/main/java/io/qameta/allure/owner/OwnerAddressParser.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright 2016-2024 Qameta Software Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.qameta.allure.owner;
+
+import org.apache.commons.validator.routines.EmailValidator;
+import org.apache.commons.validator.routines.UrlValidator;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class OwnerAddressParser {
+    private static final Pattern RFC2822_ADDRESS = Pattern.compile("^([^<>]+)\\s+<\\s*(\\S*)\\s*>$");
+
+    private OwnerAddressParser() {
+    }
+
+    public static OwnerAddress parseAddress(final String maybeAddress) {
+        if (maybeAddress == null || maybeAddress.isEmpty()) {
+            return null;
+        }
+
+        // Prevent performance degradation for plain text
+        if (!isLikelyAddress(maybeAddress)) {
```

Comment:
This thing is important to have for plain text. On 10,000 "John Doe {{counter}}" addresses it eats them in 20ms.

Without this optimization, it takes up to 200ms on my machine to parse 10,000 addresses.

I measured the performance between my previous simplistic regexp implementation and the actual implementation with Apache validators, and the impact was negligible — ±20-30ms (sometimes faster, sometimes slower), so I hope this is totally okay for you to accept this PR with industry-proven validators for emails and URLs instead of improvised ones by me.